src/main/java/com/arpnetworking/tsdcore/sinks/HttpSinkActor.java

```diff
@@ -140,13 +156,24 @@ public Receive createReceive() {
                     dispatchPending();
                 })
                 .match(PostRejected.class, rejected -> {
-                    processRejectedRequest(rejected);
-                    dispatchPending();
+                    final int attempt = rejected.getAttempt();
+                    if (attempt < _maximumRetries) {
+                        scheduleRetry(rejected.getRequest(), attempt);
+                    } else {
+                        processRejectedRequest(rejected);
+                        dispatchPending();
+                    }
                 })
                 .match(PostFailure.class, failure -> {
-                    processFailedRequest(failure);
-                    dispatchPending();
+                    final int attempt = failure.getAttempt();
+                    if (attempt < _maximumRetries) {
+                        scheduleRetry(failure.getRequest(), attempt);
+                    } else {
+                        processFailedRequest(failure);
```

Comment:
So two concerns with this:

1) We have no visibility in why requests are failing even if the operation is succeeding. Ideally, we would log say a warn when a request fails but an error when the operation fails. The message should also distinguish the difference between a request failure and an operation failure.

2) There is no visibility on the rate of request failures. Similar to the above, only logging the status code on the final attempt doesn't reveal that maybe the first K attempts were timeouts. Or maybe there were 4 5xx and then a 400.